packages/twenty-server/src/engine/api/graphql/workspace-query-runner/utils/handle-duplicate-key-error.util.ts

```diff
@@ -1,27 +1,63 @@
 import { msg } from '@lingui/core/macro';
-import { type QueryFailedError } from 'typeorm';
+
+import { type WorkspaceInternalContext } from 'src/engine/twenty-orm/interfaces/workspace-internal-context.interface';
 
 import { type FlatObjectMetadata } from 'src/engine/metadata-modules/flat-object-metadata/types/flat-object-metadata.type';
+import { type WorkspaceEntityManager } from 'src/engine/twenty-orm/entity-manager/workspace-entity-manager';
 import {
   TwentyORMException,
   TwentyORMExceptionCode,
 } from 'src/engine/twenty-orm/exceptions/twenty-orm.exception';
 
-interface PostgreSQLError extends QueryFailedError {
-  detail?: string;
+import { findConflictingRecord } from './find-conflicting-record.util';
+import {
+  parsePostgresConstraintError,
+  type PostgreSQLError,
+} from './parse-postgres-constraint-error.util';
+
+interface DuplicateKeyErrorWithMetadata extends TwentyORMException {
+  conflictingRecordId?: string;
+  conflictingObjectNameSingular?: string;
 }
 
-export const handleDuplicateKeyError = (
-  _error: PostgreSQLError,
-  _objectMetadata: FlatObjectMetadata,
-) => {
-  // Since we no longer have indexMetadatas in FlatObjectMetadata,
-  // we provide a generic error message
-  throw new TwentyORMException(
+export const handleDuplicateKeyError = async (
```

Comment:
@Weiko could you please review this PR? I made a few minor comments but I'm not sure if it's the right approach overall?